src/main/kotlin/com/bugsnag/android/gradle/internal/BugsnagHttpClientHelper.kt

```diff
@@ -25,23 +27,55 @@ abstract class BuildServiceBugsnagHttpClientHelper
 
     interface Params : BuildServiceParameters {
         val timeoutMillis: Property<Long>
+        val retryCount: Property<Int>
     }
 
-    override val okHttpClient: OkHttpClient by lazy { newClient(parameters.timeoutMillis.get()) }
+    override val okHttpClient: OkHttpClient by lazy {
+        newClient(parameters.timeoutMillis.get(), parameters.retryCount.get())
+    }
 }
 
 /** A simple instance-based [BugsnagHttpClientHelper] for use on Gradle <6.1. */
-class LegacyBugsnagHttpClientHelper(timeoutMillis: Provider<Long>) : BugsnagHttpClientHelper {
-    override val okHttpClient: OkHttpClient by lazy { newClient(timeoutMillis.get()) }
+class LegacyBugsnagHttpClientHelper(
+    timeoutMillis: Provider<Long>,
+    retryCount: Provider<Int>
+) : BugsnagHttpClientHelper {
+    override val okHttpClient: OkHttpClient by lazy { newClient(timeoutMillis.get(), retryCount.get()) }
 }
 
-internal fun newClient(timeoutMillis: Long): OkHttpClient {
+internal fun newClient(
+    timeoutMillis: Long,
+    retryCount: Int
+): OkHttpClient {
     val timeoutDuration = Duration.ofMillis(timeoutMillis)
+    val interceptor = retryInterceptor(retryCount)
     return OkHttpClient.Builder()
         .readTimeout(Duration.ZERO)
         .writeTimeout(Duration.ZERO)
         .connectTimeout(Duration.ZERO)
         .callTimeout(timeoutDuration)
+        .addInterceptor(interceptor)
         .build()
 }
 
+internal fun retryInterceptor(maxRetries: Int): Interceptor {
+    return object : Interceptor {
+        override fun intercept(chain: Interceptor.Chain): Response {
+            var response: Response? = null
+            var attempts = 1
+            var cause: Throwable? = null
+
+            while (response == null && maxRetries >= attempts) {
+                try {
+                    val request = chain.request()
+                    response = chain.proceed(request)
+                } catch (exc: Throwable) {
+                    cause = exc
+                }
+                attempts++
+            }
+            return response ?: throw IllegalStateException("Bugsnag request failed to complete", cause)
```

Comment:
I think this would be a little clearer with a do/while loop, because we always want the request to happen at least once (which is currently guaranteed with `response == null`) and then a `maxRetries` of 0 works as expected

```suggestion
            var attempts = 0
            var cause: Throwable? = null

            do {
                try {
                    val request = chain.request()

                    return chain.proceed(request)
                } catch (exc: Throwable) {
                    cause = exc
                }
                
                attempts++
            } while (attempts < maxRetries)

            throw IllegalStateException("Bugsnag request failed to complete", cause)
```

Reply:
This is clearer - I've adopted this suggestion.